kubernetes/nodecollection.go

```diff
@@ -148,48 +261,38 @@ func ensureNodeSource(config KubeAgentConfig) (KubeAgentConfig, error) {
 
 	config.NodeClient = nodeClient
 
-	nodes, err := clientSetNodeSource.GetNodes()
+	nodes, err := clientSetNodeSource.GetReadyNodes()
 	if err != nil {
 		return config, fmt.Errorf("error retrieving nodes: %s", err)
 	}
 
-	ip, port, err := clientSetNodeSource.NodeAddress(&nodes.Items[0])
+	firstNode := &nodes[0]
+
+	ip, port, err := clientSetNodeSource.NodeAddress(firstNode)
 	if err != nil {
 		return config, fmt.Errorf("error retrieving node addresses: %s", err)
 	}
```

Comment:
Not going to block this PR, but I just noticed that we're failing this branch on being unable to get the node address, also, despite that info only being used in the direct connection test. Might want to rethink that in future....